From: NIIBE Yutaka <gniibe@fsij.org>
Date: Tue, 14 May 2019 11:20:07 +0900
Subject: g10: Fix possible null dereference.

* g10/armor.c (armor_filter): Access ->d in the internal loop.

--

Cherry-picked master commit of:
	802a2aa300bad3d4385d17a2deeb0966da4e737d

GnuPG-bug-id: 4494
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
(cherry picked from commit 5b22d2c400890fc366ccb7ca74ee886d9cef22a3)
---
 g10/armor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/g10/armor.c b/g10/armor.c
index cc80968..36215a3 100644
--- a/g10/armor.c
+++ b/g10/armor.c
@@ -1156,10 +1156,10 @@ armor_filter( void *opaque, int control,
 	      }
 
 	    /* write the comment strings */
-	    for(s=comment->d;comment;comment=comment->next,s=comment->d)
+	    for(;comment;comment=comment->next)
 	      {
 		iobuf_writestr(a, "Comment: " );
-		for( ; *s; s++ )
+		for( s=comment->d; *s; s++ )
 		  {
 		    if( *s == '\n' )
 		      iobuf_writestr(a, "\\n" );
